lib/Kernel.php

```diff
@@ -167,13 +167,34 @@ public function registerContainerConfiguration(LoaderInterface $loader)
             $loader->load($systemConfigFile);
         }
 
-        foreach (['image-thumbnails', 'video-thumbnails', 'custom-reports', 'document-types', 'web-to-print', 'predefined-properties', 'predefined-asset-metadata', 'staticroutes', 'perspectives', 'custom-views'] as $configDir) {
+        foreach (['video-thumbnails', 'custom-reports', 'document-types', 'web-to-print', 'predefined-properties', 'predefined-asset-metadata', 'staticroutes', 'perspectives', 'custom-views'] as $configDir) {
             $configDir = PIMCORE_CONFIGURATION_DIRECTORY . "/$configDir/";
             if (is_dir($configDir)) {
                 // @phpstan-ignore-next-line
                 $loader->import($configDir);
             }
         }
+
+        $configArray = [
+            ['storageDirectoryEnvVariableName' => 'PIMCORE_CONFIG_STORAGE_DIR_IMAGE_THUMBNAILS',
+                'defaultStorageDirectoryName' => 'image-thumbnails']
+        ];
+
+        foreach($configArray as $config) {
+            $configDir = $config['storageDirectoryEnvVariableName'];
+            $configDir = $_ENV[$configDir] ?? null;
```

Comment:
Is `$_ENV` enough here? Or should we check `$_SERVER` too?

Reply:
`$_ENV` should be fine, because of: 
https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Dotenv/Dotenv.php#L151

Reply:
@brusch This line set the $_ENV values to $_SERVER not the other way around

Reply:
Yes, and therefore `$_SERVER` should work as well, right?

Reply:
Yes, I think we should use the $_SERVER var

Reply:
For what reason? We're using `$_ENV` already in other places as well ... 🤔

Reply:
I wouldn't rely on Symfony DotEnv, since it's not necessarily used, especially in prod the variables are probably defined otherwise.

> We're using $_ENV already in other places as well ...

Not really. You're mostly using `$_SERVER`, sometimes with a fallback to `$_ENV`. There's only two places where only `$_ENV` is used: https://github.com/pimcore/pimcore/blob/10.x/lib/Config/LocationAwareConfigRepository.php#L203 and https://github.com/pimcore/pimcore/blob/10.x/bundles/InstallBundle/Installer.php#L382-L383

Reply:
Ok, I'm fine with `$_SERVER` as well, but we should then change it in the other places where `$_ENV` is used as well. Can you provide a PR for that? 
@mcop1 Please consider 😉

Reply:
Changed it to $_SERVER